### PR TITLE
hw7+hw8 

### PR DESCRIPTION
**Pull request by Raily01**

### CODE REVIEW

File: src/components/molecules/Table/Table.jsx [hunk 1]
```diff
@@ -26,64 +29,50 @@ const StyledTd = styled.td`
...
 `;
 const Table = () => {
+  const { projects, loading } = useProjects();
+  // console.log(projects, loading);
   const test = () => {
     console.log("pupupu");
   };
+  const navigate = useNavigate();
+  const routeChange = (id1) => {
+    console.log(id1);
```
Comment (VeraKutyavina) [inline src/components/molecules/Table/Table.jsx]: please remove console.log

File: src/components/templates/DefaultTemplate.jsx [hunk 2]
```diff
@@ -25,17 +31,53 @@ const StyledTemp = styled.h1`
...
+    if (window.scrollY > 0) {
+      // console.log("scrolled");
+      setShowScrollButton(true);
+    } else {
+      setShowScrollButton(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", onScroll);
```
Comment (VeraKutyavina) [inline src/components/templates/DefaultTemplate.jsx:57]: Need to add this in useEffect
```
return () => {
      document.removeEventListener("scroll", handleScroll);
    };
```
It remove remove event listeners after component unmount

File: src/components/templates/DefaultTemplate.jsx [hunk 3]
```diff
@@ -25,17 +31,53 @@ const StyledTemp = styled.h1`
...
+
 const DefaultTemplate = ({ children }) => {
-  const test = () => {
-    console.log("pupupu");
+  const [showScrollButton, setShowScrollButton] = useState(false);
+
+  const onScroll = () => {
+    // console.log("what..");
+    if (window.scrollY > 0) {
+      // console.log("scrolled");
```
Comment (VeraKutyavina) [inline src/components/templates/DefaultTemplate.jsx]: please remove comments

File: src/components/templates/DefaultTemplate.jsx [hunk 4]
```diff
@@ -25,17 +31,53 @@ const StyledTemp = styled.h1`
...
   };
+  // const[showScrollButton, setShowScrollButton] = useState(false);
+  const navigate = useNavigate();
   return (
     <StyledCont>
       <Header />
+      <ButtonWrapperGoBack>
+        <Button onClick={() => navigate(-1)} label="Go back" />
+      </ButtonWrapperGoBack>
+      <ButtonWrapperGoBack onClick={() => navigate(-1)} />
```
Comment (VeraKutyavina) [inline src/components/templates/DefaultTemplate.jsx]: Now it render empty div in html, you can remove it

File: src/components/molecules/Table/Table.jsx [hunk 5]
```diff
@@ -26,64 +29,50 @@ const StyledTd = styled.td`
...
+            <tr>
+              <StyledTh>Name</StyledTh>
+              <StyledTh>Description</StyledTh>
+              <StyledTh>Created at</StyledTh>
+              <StyledTh>&nbsp;</StyledTh>
+              <StyledTh>&nbsp;</StyledTh>
+              <StyledTh>&nbsp;</StyledTh>
+            </tr>
+            {projects.map(({ name, description, createdAt, id }) => (
+              <tr>
```
Comment (DmitriiMiasnikov) [inline src/components/molecules/Table/Table.jsx]: add prop key={id} to <tr>, cause it is list item
Screenshot 2023-04-24 at 15 52 46

File: src/components/molecules/Table/Table.jsx [hunk 6]
```diff
@@ -26,64 +29,50 @@ const StyledTd = styled.td`
...
+                </StyledTd>
+                <StyledTd>
+                  <Button label="Destroy" onClick={test} color="thistle" />
+                </StyledTd>
+              </tr>
+            ))}
+          </tbody>
+        </StyledTable>
+      )}
+      <div>{loading && <h1>LOADING NOW</h1>}</div>
```
Comment (DmitriiMiasnikov) [inline src/components/molecules/Table/Table.jsx]: div unneeded, it shouldn't render after loading
```suggestion
      {loading && <h1>LOADING NOW</h1>}
```
Screenshot 2023-04-24 at 15 54 45

**Review by VeraKutyavina — APPROVED** [review]

**Review by DmitriiMiasnikov — APPROVED** [review]

**Pull request merged**